cadc-util/src/main/java/ca/nrc/cadc/db/version/KeyValueDAO.java

```diff
@@ -197,10 +193,24 @@ public void delete(String name) {
         jdbc.update(sql, arg);
     }
 
+    /**
+     * List all the KeyValue objects in the database.
+     *
+     * @return List of KeyValue objects.
+     * @throws org.springframework.dao.DataAccessException if there is a problem querying the database.
+     */
+    public List<KeyValue> list() {
+
+        SelectStatementCreator sel = new SelectStatementCreator();
+        return jdbc.query(sel, new KeyValueRowMapper());
+    }
+
     private class SelectStatementCreator implements PreparedStatementCreator {
 
         private final boolean forUpdate;
         private String model;
+        private String name;
```

Comment:
This looks wrong in several ways.

1. setWhere() is never called and should be removed;
2. the `name` column in the db is the only query constraint (when not listing)
3. listing (model == null) and locking (`forUpdate`) should never be used together

There is no need for both `model` and `name` fields... `model` makes less sense now so should be dropped in favour of `name` (eg in `setValue(String name)` and in `loadValues()`). 

The conditional in creating SQL would then be correct, but the SQL itself would be wrong: the column name is literally "name", not the value of name.

for 3, we need a single method that takes the `name` and `forUpdate` value... so that implies something like `setValue(String, boolean)`

Reply:
Updated for the above comments.